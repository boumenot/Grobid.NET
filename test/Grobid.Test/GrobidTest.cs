using System;
using System.IO;

using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Xunit;

using Grobid.NET;
using org.apache.log4j;
using org.grobid.core;

namespace Grobid.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class GrobidTest
    {
        static GrobidTest()
        {
            // BasicConfigurator.configure();
            // org.apache.log4j.Logger.getRootLogger().setLevel(Level.DEBUG);
        }

        [Fact]
        [Trait("Test", "EndToEnd")]
        public void ExtractTest()
        {
            var binPath = Environment.GetEnvironmentVariable("PDFTOXMLEXE");

            var factory = new GrobidFactory(
                "grobid.zip",
                binPath,
                Directory.GetCurrentDirectory());

            var grobid = factory.Create();
            var result = grobid.Extract(@"essence-linq.pdf");

            result.Should().NotBeEmpty();
            Approvals.Verify(result);
        }

        [Fact]
        public void Test()
        {
            var x = GrobidModels.NAMES_HEADER;
            x.name().Should().Be("NAMES_HEADER");
            x.toString().Should().Be("name/header");
        }
    }
}
