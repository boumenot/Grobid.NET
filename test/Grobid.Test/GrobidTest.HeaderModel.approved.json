{
    "Title": "The essence of language-integrated query",
    "Abstract": "Language-integrated query is receiving renewed attention, in part because of its support through Microsoft ' s LINQ framework. We present a theory of language-integrated query based on quotation and normalisation of quoted terms. Our technique supports-tion over values and predicates, composition of queries, dynamic generation of queries, and queries with nested intermediate data. Higher-order features prove useful even for constructing first-order queries. We prove that normalisation always succeeds in translat-ing any query of flat relation type to SQL. We present experimental results confirming our technique works, even in situations where Microsoft ' s LINQ framework either fails to produce an SQL query or, in one case, produces an avalanche of SQL queries. Computing Classification Software and its engineering! Soft-ware notations and tools! General programming languages! Language features",
    "Authors": [
        {
            "Name": "James Cheney",
            "Affiliation": "The University of Edinburgh",
            "EMail": "jcheney@inf.ed.ac.uk"
        },
        {
            "Name": "Sam Lindley",
            "Affiliation": "University of Strathclyde",
            "EMail": "sam.lindley@strath.ac.uk"
        },
        {
            "Name": "Philip Wadler",
            "Affiliation": "The University of Edinburgh",
            "EMail": "wadler@inf.ed.ac.uk"
        }
    ],
    "Keywords": [
        "lambda calculus",
        "LINQ",
        "F#",
        "quotation",
        "anti-quotation"
    ]
}